prune.go

```diff
@@ -266,6 +284,15 @@ func dirStats(dir string) (*Stats, error) {
 	return &stats, err
 }
 
+// compile an regular expression array
```

Comment:
Add an function to compile the RegExp.

Just run once at program run

We do not need to compile patten every times